Assets/GameLogic/StoreLogicService/StoreLogicService.cs

```diff
@@ -10,20 +10,41 @@ public interface IStoreLogicService
         IDataLayer DataLayer { get; set; }
         void OnAppStorePurchaseSucceeded(string productId);
         Action<string> LogToDebugOutput { get; set; }
+        List<string> ValidateProducts(List<ProductInfo> products);
     }
 
     public class StoreLogicService : IStoreLogicService
     {
         private static readonly StoreLogicService _instance = new StoreLogicService();
 
+        private ProductInfo FindSmallestProductInfo (List<ProductInfo>productInfos)
+        {
+              return null;
+        }
+
         private decimal CalculateSavePercent(ProductInfo smallPackage, ProductInfo bulkPackage)
         {
+            decimal standardPrice = smallPackage.Price;
+            decimal bulkPrice = standardPrice * bulkPackage.Price;
+            decimal packagePrice = bulkPrice / bulkPackage.Quantity;
+            decimal priceSaved = (1 - packagePrice) * 100;
+            return priceSaved;
+            
             throw new NotImplementedException();
         }
 
         private string GenerateSavePctString(decimal savePct)
-        {
-            throw new NotImplementedException();
+        {                           
+            if (savePct <= 0) 
+            {
+             return string.Empty;
```

Comment:
could you please make this a single point of return?

The if/else statements will assign a value to a local var, then we'll return that var at the end.